src/AspNet.Security.OAuth.VkId/VkIdAuthenticationHandler.cs

```diff
@@ -0,0 +1,337 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System.Globalization;
+using System.Net;
+using System.Net.Http.Headers;
+using System.Security.Claims;
+using System.Security.Cryptography;
+using System.Text;
+using System.Text.Encodings.Web;
+using System.Text.Json;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Primitives;
+using Base64UrlEncoder = Microsoft.AspNetCore.Authentication.Base64UrlTextEncoder;
+
+namespace AspNet.Security.OAuth.VkId;
+
+public sealed partial class VkIdAuthenticationHandler : OAuthHandler<VkIdAuthenticationOptions>
+{
+    public VkIdAuthenticationHandler(
+        [NotNull] IOptionsMonitor<VkIdAuthenticationOptions> options,
+        [NotNull] ILoggerFactory logger,
+        [NotNull] UrlEncoder encoder)
+        : base(options, logger, encoder)
+    {
+    }
+
+    protected override string BuildChallengeUrl(
+        [NotNull] AuthenticationProperties properties,
+        [NotNull] string redirectUri)
+    {
+        var parameter = Options.Scope;
```

Comment:
Looks like these temp variables aren't very useful: feel free to remove them and do the formatting directly in the dictionary initializer to save a few lines.